### DISCUSSION TITLE
Can you assign White Matter, CSF masks,etc. during the confound correction step instead of preprocessing?

### DISCUSSION THREAD

**Discussion by geowk** [CoBrALab/RABIES#354] [General]
Hello,

We have a dataset preprocessed with Rabies, but wanted to change the CSF mask with an new eroded version.

Instead of re-running preprocessing, is there a way to add the new eroded CSF mask in the confound correction command step to run CSF signal regression or ica-aroma?

Thanks.

g

**Comment by Gab-D-G** [comment]
Hi,

Unfortunately it is not possible to change the masks without re-running preprocessing, as the masks are resampled onto the preprocessed timeseries at this stage. However, if the exact same preprocessing command is run, but only the paths to the CSF/WM masks is changed, then only the last stages of preprocessing should be run, which should be quick. Do not change the brain mask however, as this file is used in early stages of the pipeline.

Let me know if there are issues encountered with re-running preprocessing.

Gabriel

**Comment by geowk** [comment]
Hello,

Thank you for your response.

I tried running it by changing the path of the CSF and WM masks.

I received the following message:

```
Traceback (most recent call last):
  File "/home/rabies/miniforge/bin/rabies", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/rabies/RABIES/scripts/rabies", line 3, in <module>
    execute_workflow()
  File "/home/rabies/RABIES/rabies/run_main.py", line 28, in execute_workflow
    log = prep_logging(opts, output_folder)
  File "/home/rabies/RABIES/rabies/run_main.py", line 72, in prep_logging
    raise ValueError(f"""
ValueError: 


            A previous run was indicated by the presence of /preprocess_outputs/rabies_preprocess.pkl.
            This can lead to inconsistencies between previous outputs and the log files.
            To prevent this, you are required to manually remove /preprocess_outputs/rabies_preprocess.pkl, and we 
            recommend also removing previous datasinks from the preprocess RABIES step.
```

Is it just simply changing the path of the CSF and WM?

Do I change this in the rabies_preprocess.log file or within the rabies command line for preprocessing?

I looked within the original preprocessing folder and it looked like nothing had been modified. I received the mentioned message written to screen and then it eventually went into confound correction and analysis steps. However, I think it still used the original WM and CSF masks since nothing had been modified within the preprocessing folder that I could tell.

g

**Reply by Gab-D-G** [reply]
"you are required to manually remove /preprocess_outputs/rabies_preprocess.pkl" - by default RABIES prevents you from removing old outputs by detecting this file. If you remove it, you'll be able to run the pipeline properly. It may be better to also remove the datasink folders to make sure you are not going mix old outputs with new ones.

**Comment by geowk** [comment]
Okay, thanks.

When you say remove the datasink folders, do you mean just the bold_datasink, confounds, anat, and transforms_datasink folders in the preprocess folder. I also noticed there are some *_datasink subfolders within the main_wf/ folder.

For example:

main_wf/_scan_info_subject_id0001.session01_split_name_sub-0001_ses-01_desc-o_T2w/_run_None/bold_datasink

Should I get rid of those subfolders as well?

-g

**Reply by Gab-D-G** [reply]
I indeed only mean the datasink folder within the top directory, no need to go into main_wf.

**Comment by geowk** [comment]
Great, Gabe. Thanks for the clarification.

Before your reply, I tried it with just removing the .pkl file and it worked. Modified the CSF and WM masks, but everything else it went through pretty quick and then on to confound_correction and then analysis.

Tried it again a copy of the dataset with also removing the top directory datasink folders, and it seems to be working as well.

Thanks again for your help. Saved a lot of time. 

-g

**Reply by Gab-D-G** [reply]
Glad to hear this helps! Let me know if you need further assistance.
